### PR TITLE
Use self-hosted runner for test suite

### PR DESCRIPTION
**Pull request by atomaka**

### CODE REVIEW

**Comment by imstar15** [comment]
Excuse me, should we keep the test-coverage?

**Comment by atomaka** [comment]
> Excuse me, should we keep the test-coverage?

Yup! We should. I was just wanted to get a basic test build passing so that I could benchmark against some self-hosted nodes I'm going to test.

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -1,57 +1,39 @@
...
-      - name: Test coverage
-        run: |
-          sh ./.maintain/test-coverage/test-coverage.sh
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Setup Rust
+      run: rustup show
+    - name: Build
+      run: |
+        cargo build --release --all
```
Comment (nvengal) [inline .github/workflows/check.yml:36]:
```suggestion
        cargo build --release --locked --all-features
```
We may want to consider further limiting this to exclude the runtime-benchmarks features and anything else that might be low value but time consuming. But this is a good start regardless.

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -1,57 +1,39 @@
...
+    - name: Test
+      run: |
+        cargo test --release --all
```
Comment (nvengal) [inline .github/workflows/check.yml:39]:
```suggestion
        cargo test --release --locked --all-features
```

File: rust-toolchain.toml [hunk 3]
```diff
@@ -1,3 +1,5 @@
 [toolchain]
 channel = "nightly-2021-12-15"
+components = ["rustfmt"]
 targets = ["wasm32-unknown-unknown"]
+profile = "minimal"
```
Comment (nvengal) [inline rust-toolchain.toml:5]: I don't fully know the implications of this. Looks like minimal is the right choice for CI but setting it here will also affect our local development.

Could explicitly set the profile for CI in the action
```
rustup set profile minimal
```

Comment (nvengal) [reply rust-toolchain.toml:5]: Was referencing this doc- https://rust-lang.github.io/rustup/concepts/profiles.html

**Pull request merged**